js32.cpp

```diff
@@ -45,9 +46,16 @@ JSScript* JS_CompileFile(JSContext* cx, JSObject* globalObject, std::string file
         str[2] = ' ';
     }
 
-    JSScript* rval = JS_CompileScript(cx, globalObject, str.c_str(), str.size(), fileName.c_str(), 1);
-    JS_AddNamedScriptRoot(cx, &rval, fileName.c_str());
+    char* nFileName = UnicodeToAnsi(fileName.c_str());
+
+    JS::RootedObject obj(cx, globalObject);

```

Comment:
Curious what this addition accomplishes. Otherwise, My code review is complete! :)

Reply:
It sets the compile mode to utf8 allowing variable names to be in other languages